libethereum/BlockChain.cpp

```diff
@@ -1159,41 +1084,42 @@ void BlockChain::rewind(unsigned _newHead)
 
 tuple<h256s, h256, unsigned> BlockChain::treeRoute(h256 const& _from, h256 const& _to, bool _common, bool _pre, bool _post) const
 {
-//	cdebug << "treeRoute" << _from << "..." << _to;
 	if (!_from || !_to)
 		return make_tuple(h256s(), h256(), 0);
+
+	BlockDetails const fromDetails = details(_from);
+	BlockDetails const toDetails = details(_to);
+	if (fromDetails.isNull() || toDetails.isNull())
```

Comment:
This is needed to handle a special case when the parent of inserted block is not present in DB